[Breaking] Remove deprecated built-in AMP

devjiwonchoi opened the pull request:
This PR removes the deprecated built-in AMP support. Will follow up with cleaning up the examples, docs, and codemods.

devjiwonchoi committed:
Remove tests and fixtures from test/ dir

devjiwonchoi commented:
* **#84312** <a href="https://app.graphite.dev/github/pr/vercel/next.js/84312?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/vercel/next.js/84312?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

ijjk commented:
## Failing test suites <!--46461342544-->
Commit: f3324b13397e35b030e3250dcf13b3263aab158f | [About building and testing Next.js](https://github.com/vercel/next.js/blob/canary/contributing.md)

`pnpm test test/integration/next-image-new/default/test/static.test.ts` (turbopack)
- Static Image Component Tests > development mode > Should allow an image with a static src to omit height and width ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f3324b13397e35b030e3250dcf13b3263aab158f%22+%40test.name%3A%22Static+Image+Component+Tests+development+mode+Should+allow+an+image+with+a+static+src+to+omit+height+and+width%22+%40test.type%3A%22turbopack%22+%40test.status%3A%22fail%22))

<details>
<summary>Expand output</summary>

  ● Static Image Component Tests › development mode › Should allow an image with a static src to omit height and width

    page.waitForSelector: Timeout 5000ms exceeded.
    Call log:
      - waiting for locator('#basic-static') to be visible

      511 |
      512 |     return this.startChain(async () => {
    > 513 |       const el = await page.waitForSelector(selector, {
          |                             ^
      514 |         timeout,
      515 |         state,
      516 |       })

      at waitForSelector (lib/browsers/playwright.ts:513:29)
      at Playwright._chain (lib/browsers/playwright.ts:643:23)
      at Playwright._chain [as startChain] (lib/browsers/playwright.ts:624:17)
      at Playwright.startChain [as waitForElementByCss] (lib/browsers/playwright.ts:512:17)
      at Playwright.waitForElementByCss [as elementByCss] (lib/browsers/playwright.ts:397:17)
      at Playwright.elementByCss [as elementById] (lib/browsers/playwright.ts:417:17)
      at Object.elementById (integration/next-image-new/default/test/static.test.ts:26:26)

</details>

devjiwonchoi committed:
Remove "deletable" files and modify related files
Related files are that imports the deleted files or has relation to those files

devjiwonchoi committed:
Remove AMP sources

devjiwonchoi committed:
Remove AMP dependencies

- Remove deps
- Remove amp files from packge.json "files"
- Remove from taskfile.js
- Run ncc-compiled

devjiwonchoi committed:
Clean up AMP related repo sources

devjiwonchoi commented on packages/next/src/shared/lib/side-effect.tsx:
```diff
@@ -10,7 +10,6 @@ export type SideEffectProps = {
   ) => State
   handleStateChange?: (state: State) => void
   headManager: any
-  inAmpMode?: boolean
```
Good catch

devjiwonchoi commented:
The deployment [test failure](https://github.com/vercel/next.js/actions/runs/18081154649/job/51445760043?pr=84312) of `test/e2e/async-modules/index.test.ts` is irrelevant to this PR.
Confirmed in https://github.com/vercel/next.js/pull/84349, and will follow up fixing it.
